framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -184,7 +184,6 @@ public AnnotatedTypeMirror visitMemberSelect(MemberSelectTree node,
         if (elt.getKind().isClass() || elt.getKind().isInterface())
             return f.fromElement(elt);
 
-        // The expression might be a primitive type (as in "int.class").
         if (!(node.getExpression() instanceof PrimitiveTypeTree)) {
```

Comment:
I would remove the negation here and add the code from the else branch here.
Then document what case the current then-branch (the new else-branch) handles.